### PR TITLE
grpc: switch to new scheme

### PR DESCRIPTION
**Pull request by sougou**
This is a work-around for: https://github.com/grpc/grpc-go/issues/1633.

Commit 1
```
1.7.1 has a fix for the case where a request would fast-fail if a
connection is not ready yet. This behavior is needed for the next
change where we're going to change all our dials to be non-blocking.
```

Commit 2
```
grpc is moving towards error-less dialing. Failures can happen
only during requests. However, all requests currently fail with
'Unavailable' error due to a design constraint. There is currently
no way to programmatically get to the underlying error.

Due to this, we have to rely on grpc's logging to find the root
cause of failures. This is now turned on for all programs that
use the grpc client.

For now, I'm leaving the dial timeout parameters in. They'll just
be ignored. Once we're confident that this approach works, we
can clean them all up.
```

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -122,7 +122,7 @@ ln -snf $consul_dist/consul $VTROOT/bin/consul
 # install gRPC C++ base, so we can install the python adapters.
 # this also installs protobufs
 grpc_dist=$VTROOT/dist/grpc
-grpc_ver=v1.0.0
+grpc_ver=v1.7.0
```
Comment (alainjobart) [inline bootstrap.sh]: 1.7.1?

File: travis/install_grpc.sh [hunk 2]
```diff
@@ -51,5 +51,5 @@ fi
 
 # Install gRPC python libraries from PyPI.
 # Dependencies like protobuf python will be installed automatically.
-grpcio_ver=1.6.0
+grpcio_ver=1.7.0
```
Comment (alainjobart) [inline travis/install_grpc.sh]: 1.7.1 here too, but they may not have it yet?

Comment (sougou) [reply travis/install_grpc.sh]: They don't have 1.7.1 yet.

File: go/vt/mysqlctl/grpcmysqlctlclient/client.go [hunk 3]
```diff
@@ -55,37 +57,66 @@ func factory(network, addr string, dialTimeout time.Duration) (mysqlctlclient.My
...
+func (c *client) withRetry(ctx context.Context, f func() error) error {
+	var lastError error
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("%v: %v", ctx.Err(), lastError)
+		default:
+		}
+		if err := f(); err != nil {
+			if grpc.Code(err) == codes.Unavailable {
```
Comment (rajatgoel) [inline go/vt/mysqlctl/grpcmysqlctlclient/client.go]: lastError = err?

**Pull request merged**